### ISSUE TITLE
Schlangenkopf lag on any setting. 

### ISSUE THREAD

**Issue by Wolfenburger** [Realm667/WolfenDoom#452]
I know the comment will be "git a better pc" I run doom eternal at ultranightmare with 60fps. I can run almost any game. However maybe the game just doesn't click with my chip set. Which can be a reality. But like I can be standing still or panning the camera at different intervals and the map will explode with lag, BUT at random. Never in the same place. It comes and goes. Like I could be standing in a building and the lag explodes. With nothing moving but than it clears out. Than explodes moments later, but not at the same area it was experienced. Unsure why it does. My friend that plays the public release aka the one on moddb also experienced the same issues. This is free to be closed whenever. I just want the game to be awesome.

**Comment by Tormentor667** [comment]
Thanks for the hint, but it is hard to track that down when it is not reproducable at certain circumstances.

**Comment by Wolfenburger** [comment]
> Thanks for the hint, but it is hard to track that down when it is not reproducable at certain circumstances.

I could probably record it. It's just very odd. Also your computer built the game. So you're undoubtedly not going to have optimization issues. Ill see if I can record it later.

**Comment by Tormentor667** [comment]
Well, the technique is the same for everyone so it doesn't matter if it was built by my pc or any other pc ;) We didn't get issue reports about C2M3 from anyone else of our testers - so it could just be your pc either.

Anyway, we need some hints to track it down.

**Comment by Username-N00b-is-not-available** [comment]
Hints and/or videos would definitely be appreciated. It is also important to know whether the FPS dropdown renders the game unplayable or not. Do you have uncapped framerate BTW?

**Comment by Wolfenburger** [comment]
> Hints and/or videos would definitely be appreciated. It is also important to know whether the FPS dropdown renders the game unplayable or not. Do you have uncapped framerate BTW?

I do believe I have a uncapped rate

**Comment by Talon1024** [comment]
The GZDoom developers are testing a new rendering backend: https://forum.zdoom.org/viewtopic.php?f=49&t=71889

Perhaps that may be of interest to you.

**Comment by Wolfenburger** [comment]
> The GZDoom developers are testing a new rendering backend: https://forum.zdoom.org/viewtopic.php?f=49&t=71889
> 
> Perhaps that may be of interest to you.

Interesting. I'll get recorded footage later tonight on the lag. I'm having bad aches atm from my vaccine and don't even have the motivation to sweep

**Comment by x3003** [comment]
Overhere there are 3 major slowdowns in this level.
1, when you go into the ruins of the burned down building.
2, when you go down the stairs of the church.
3, strangely, when you are in the main (big) room of the building w/ the purple doors. Slows down a lot.

**Comment by Wolfenburger** [comment]
As you can tell this quality isn't that great because I used obs studios. Which I think I could make quality better but I don't know how. And probably shirts in the setting somewhere. But I recorded at 60 FPS. The lag in the houses you'll notice it slightly gets chuggy in the staircases in the ladders.  Also when I approached the purple door HQ that place has some explosive lag that even the crappiness of obs studios can pick up.

https://www.youtube.com/watch?v=ET8yVXjij88&ab_channel=noyou

**Comment by x3003** [comment]
>https://www.youtube.com/watch?v=ET8yVXjij88&ab_channel=noyou

Must login to confirm age. Since when are boa vids age restricted ?

**Comment by Wolfenburger** [comment]
> > https://www.youtube.com/watch?v=ET8yVXjij88&ab_channel=noyou
> 
> Must login to confirm age. Since when are boa vids age restricted ?

Blood and violence. That kinda is how. YouTube has become pretty bad these days

**Comment by Wolfenburger** [comment]
> Well, the technique is the same for everyone so it doesn't matter if it was built by my pc or any other pc ;) We didn't get issue reports about C2M3 from anyone else of our testers - so it could just be your pc either.
> 
> Anyway, we need some hints to track it down.

Actually it does matter for your PC. ;) and I doubt you on people not getting lag. Unless they outright just felt it was a accepted amount of lag. I'll literally post a picture of my computer internals. It's probably got more in than 90% of the people on this thread. Spend thousands upgrading her. Again while saying its just my computer. It may very well be, because every chip set and computer reacts differently to games. My friend that has a weaker system runs Read Dead 2 without lagging. Yet people with better computers than him lag. Also people did complain about lag on the public game. Moddb page itself has comments on it. I do believe someone said they optimized one of the chapters better. And YouTube comments as well. Your scope is to narrow. Mere testers aren't enough. There is a problem here and because you can't see it doesn't mean that it isn't here. Nice try. ;)

**Comment by Tormentor667** [comment]
First of all thanks kindly for your video, I just watched it and the laggy areas are already known to us. The problem here is the large outdoor area with a lot of sectors and visible architecture. We improved the map already as much as we could (splitting it up, hiding areas, adding sight-blocking lines) but without completely removing or rebuilding the area (and therefore destroying the atmosphere), our hands are tied here. 

The lag fortunately isn't that bad, so we simply hope that players will accept that.

**Comment by AFADoomer** [comment]
Arrogance won't fix your problems...  

I don't know what's going on with the ladders and stairs areas for you, but the purple door area is a known laggy spot.  Performance there is far better than it used to be, but some people still have problems there because of how much of the map is being rendered in this one spot.  Because you can see almost all of the way back across the map in two directions, the engine itself causes the lag.  

Your original issue said that these lag spikes happen "at random", yet you posted a video with specific problems.  Do they really happen in random locations, or are your problems repeatable?

**Comment by Hitmang468** [comment]
Made this account just for my friend. I've been having the same issues. Also screw you afadoom for thumbs downing my boi. He just was calling someone out.

**Comment by AFADoomer** [comment]
He gave himself a thumbs-up and was being an ass.

**Comment by Wolfenburger** [comment]
> Arrogance won't fix your problems...
> 
> I don't know what's going on with the ladders and stairs areas for you, but the purple door area is a known laggy spot. Performance there is far better than it used to be, but some people still have problems there because of how much of the map is being rendered in this one spot. Because you can see almost all of the way back across the map in two directions, the engine itself causes the lag.
> 
> Your original issue said that these lag spikes happen "at random", yet you posted a video with specific problems. Do they really happen in random locations, or are your problems repeatable?
> 
> I ask this because if it really is random, it could be some of the actor culling code at fault. If not, it's mapping and engine rendering that we need to worry about.

There was choppy bits as I was moving through the homes. What in my post was arrogant? If anything you're arrogant. You're coping by claiming that these spots "always" were lagging. But when I called it out prior you were claiming this crap didn't exist. And thanks for the thumbs down it was a call out on the fact someone got a thumbs up for simply blaming my PC

**Comment by AFADoomer** [comment]
Or maybe the thumbs up was for the part where he asked for more hints to track down the bug, since that was what the person who gave the thumbs up asked for in their next reply?

For the choppiness in the homes, can you tell if it's happening when you are looking in a particular direction, or in specific locations?

**Comment by Tormentor667** [comment]
@Wolfenburger and @Hitmang468 please stay polite. We are all here for the same reason, we want to improve the mod and turn it into the best possible experience.

**Comment by AFADoomer** [comment]
@Tormentor667 - I'm not helping matters.  Sorry.

**Comment by Tormentor667** [comment]
@AFADoomer all fine, I think this is a engine/mapping issue and I see no way
to further improve the situation.

**Comment by Wolfenburger** [comment]
> @Tormentor667 - I'm not helping matters. Sorry.

Look, I love the mod. I want it to work fine. But were operating completely experimentally. 100% So there isn't much to do. I'm not the brightest person but it is what it is. I just want this stuff to be fixed.

**Comment by Username-N00b-is-not-available** [comment]
@Wolfenburger:
We've already heard a few times from you that 90% of the peasant developers from our team cannot afford to buy a $3000 computer. It's all fine; who wouldn't brag about it if they had it (I probably would ;-) ).
What bothers me (and us) is that when you get presented with an opinion that does not align with yours, instead of a discussion you start attacking the developers; today is not the first time something like this had happened. The developer team consists of cultured, mature people, who would rather opt out of the discussion (or post apologies), rather than appropriately respond back to such attacks. Over the Internet, it is very easy to be rude to people.
You do not have to win a comment war here, simply because one is not going on. The ultimate decision on the changes is done by the team, and there is no use in provoking us with sharp phrases like

> Your scope is WAY to narrow.

> Lmao the cult loyalty is real here.

> Was it just to dismiss me and other people's thoughts or what?

when both you and we know they are baseless.
When it comes to contributions, your feedback is very valuable. You are a skilled player and a thorough tester, and we make great use of your reports. The only request from us is for you to behave decently, please. You are in a public place, and anonymity does not mean that you can desecrate the authority of people who are twice as old as you. Well, you have the option to, but do not expect them to listen to your opinion after that.

**Comment by Wolfenburger** [comment]
> @Wolfenburger:
> We've already heard a few times from you that 90% of the peasant developers from our team cannot afford to buy a $3000 computer. It's all fine; who wouldn't brag about it if they had it (I probably would ;-) ).
> What bothers me (and us) is that when you get presented with an opinion that does not align with yours, instead of a discussion you start attacking the developers; today is not the first time something like this had happened. The developer team consists of cultured, mature people, who would rather opt out of the discussion (or post apologies), rather than appropriately respond back to such attacks. Over the Internet, it is very easy to be rude to people.
> You do not have to win a comment war here, simply because one is not going on. The ultimate decision on the changes is done by the team, and there is no use in provoking us with sharp phrases like
> 
> > Your scope is WAY to narrow.
> 
> > Lmao the cult loyalty is real here.
> 
> > Was it just to dismiss me and other people's thoughts or what?
> 
> when both you and we know they are baseless.
> When it comes to contributions, your feedback is very valuable. You are a skilled player and a thorough tester, and we make great use of your reports. The only request from us is for you to behave decently, please. You are in a public place, and anonymity does not mean that you can desecrate the authority of people who are twice as old as you. Well, you have the option to, but do not expect them to listen to your opinion after that.

Peak projection. Peak. I was not in any way being rude. Yes your scope is narrow if you're going off a few testers. Sometimes the scope has to be widened. Also 

>We've already heard a few times from you that 90% of the peasant developers from our team cannot afford to buy a $3000 computer. It's all fine; who wouldn't brag about it if they had it (I probably would ;-) ).
Projection. Peak. I never called you peasants. You are gaslighting so hard right now it's actually scary. I simply state the computer specs because I shouldn't be having issues. Last time I complained about graphics I was given a "gg scrub your pc sux" type reply. Cope

**Comment by Username-N00b-is-not-available** [comment]
> You're probably 8 years older than me.

Yeah, this seems just about right. To clear matters here, in that sentence I was talking about the rest of the team, not about myself.

***

I am not going to give you any respect anymore, or engage in further discussions with you, on this topic or otherwise.  Not until you learn to talk to people.

**Comment by Wolfenburger** [comment]
> > You're probably 8 years older than me.
> 
> Yeah, this seems just about right. To clear matters here, in that sentence I was talking about the rest of the team, not about myself.
> 
> I am not going to give you any respect anymore, or engage in further discussions with you, on this topic or otherwise. Not until you learn to talk to people.

I'm not much younger than Dan. I'm old myself. My response will remain the same until you stop gaslighting people. "Gg your computer sux" is not a answer buddy. You just avoided that entire topic. I'm probably older than you. From my background I genuinely do not tolerate ignorance. Even my friend saw genuine rudeness in the original post. Get out of here. I'm not kissing your feet because you're a part of the team. Learn respect yourself and don't come down on others. Afa was right it was likely over him just wanting more hints. Which seems to of spawned misunderstanding. You sir are jumping in when me, afa and Tormentor were fine. You come in here with your God complex of speaking.

**Comment by AFADoomer** [comment]
Let's put this conversation on hold, please.  We are all taking this too personally and communicating poorly on all sides.
